Documentation/dev/logging.md

```diff
@@ -0,0 +1,20 @@
+# Logging Conventions
+
+We use [capnslog](capnslog) in etcd codebase for leveled logging. The following conventions for the log levels to use. 
+
+  * Error: Data has been lost, a request has failed for a bad reason, or a required resource has been lost
+    * fail to allocate disk space for WAL
+  * Warning: (Hopefully) Temporary conditions that may cause errors, but may work fine. A replica disappearing (that may reconnect) is a warning.
+    * fail to send raft messages to a remote peer
```

Comment:
* Examples:
  * Failure to send raft message to a remote peer
  * Failure to recieve heartbeat message within the configured election timeout